File: src/foam/core/Validator.java
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

package foam.core;

public interface Validator {
  public void validate(X x, foam.core.FObject obj) throws IllegalStateException;
}

File: src/foam/core/Validator.js
/**
 * @license
 * Copyright 2020 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

foam.INTERFACE({
  package: 'foam.core',
  name: 'Validator',
  methods: [
    {
      name: 'validate',
      type: 'Void',
      javaThrows: [ 'IllegalStateException' ],
      args: [
        {
          name: 'x',
          type: 'foam.core.X'
        },
        {
          name: 'obj',
          type: 'foam.core.FObject'
        }
      ]
    }
  ]
})
